manifests/init.pp

```diff
@@ -149,18 +149,30 @@
   # TODO: doesn't allow deploying a Pulp non-mirror without Foreman
   $shared_with_foreman_vhost = !$pulpcore_mirror
 
+  if $shared_with_foreman_vhost {
+    include foreman::config::apache
+    Class['foreman::config::apache'] -> Class['pulpcore::apache']
+    $servername = $foreman::config::apache::servername
+  } else {
+    include certs::apache
+    Class['certs::apache'] ~> Class['pulpcore::apache']
+    $servername = $certs::apache::hostname
```

Comment:
I think this is true in all cases and could be pulled out of the if-else. I realize we did not do this before but we could have.

Reply:
Which part specifically -- the include and the ordering relationship? Shouldn't that be causing a problem currently if it's required to be defined outside of the else?

Pulling the `$servername = ...` out of the else here would cause a problem, because it is defined differently under the if.

Reply:
I did find that we `include certs::apache` in manifests/reverse_proxy.pp; and `foreman_proxy_content::reverse_proxy` will be used when `$pulpcore_mirror or $reverse_proxy`, meanwhile we do explicitly include it here when `$shared_with_foreman_vhost` (i.e. when `!$pulpcore_mirror`).

So since it gets ultimately included when `$pulpcore_mirror or !$pulpcore_mirror`, that is always. I think I see your point that it could be included at the top level here. And that passes the sanity test -- when do we not need apache and its certs?

But what if the conditions change in the future? For example if we no longer have `$shared_with_foreman_vhost = !$pulpcore_mirror` but instead some other logic? I think I prefer to let puppet handle that thinking for us instead of opening up the possibility of always including something we may no longer need in the future?

Reply:
I am most saying, that for the forseeable future, and today, the servername where we deploy Apache is always mapped to `    $servername = $certs::apache::hostname` and in some ways is the best source of truth for this value given these are the certificates under which Apache is running.

Reply:
> I am most saying, that for the forseeable future, and today, the servername where we deploy Apache is always mapped to ` $servername = $certs::apache::hostname` and in some ways is the best source of truth for this value given these are the certificates under which Apache is running.

This is also what I always used as leading. While technically there are `cnames` and `subjectAltName`, we can also assume a `commonName` in our certificates and assume that works.

Technically you could also do it the other way around and have a `$servername` parameter and call `class { 'certs::apache': hostname => $servername }` but that's effectively the same thing.